### PR TITLE
DEV2-2591 fix environment variables

### PR DESCRIPTION
**Pull request by dimacodota**
DEV2-2591 fix environment variables

### CODE REVIEW

File: src/main/java/com/tabnine/binary/BinaryProcessGateway.kt [hunk 1]
```diff
@@ -54,13 +61,16 @@ open class BinaryProcessGateway {
...
     }
 
-    private fun inheritIDEProxySettings(env: MutableMap<String, String>) {
-        val proxyString = "${HttpConfigurable.getInstance().PROXY_HOST}:${HttpConfigurable.getInstance().PROXY_PORT}"
-        if (proxyString.isNotBlank()) {
-            env["HTTPS_PROXY"] = proxyString
-            env["https_proxy"] = proxyString
-            env["HTTP_PROXY"] = proxyString
-            env["http_proxy"] = proxyString
+    private fun inheritIDEProxySettings(env: MutableMap<String, String>, httpConfigurable: HttpConfigurable) {
```
Comment (amircodota) [inline src/main/java/com/tabnine/binary/BinaryProcessGateway.kt:64]: isn't there user/pass there too?

**Review by amircodota — COMMENTED** [review]
see my comments

**Comment by amircodota** [comment]
@dimacodota Do you have a way to test this?

File: src/main/java/com/tabnine/binary/BinaryProcessGateway.kt [hunk 2]
```diff
@@ -20,48 +18,51 @@ open class BinaryProcessGateway {
...
 
-    private fun setNoProxyForTabnineServerUrl(
-        serverUrl: Optional<String>,
-        env: MutableMap<String, String>
-    ) {
-        val serverUrlURL = URL(serverUrl.get())
-        val tabnineServerProxy = CommonProxy.getInstance().select(serverUrlURL).firstOrNull()
-        if (tabnineServerProxy?.type() == Proxy.Type.DIRECT) {
+        if (!serverUrl.isEmpty) {
+            val serverUrlURL = URL(serverUrl.get())
```
Comment (amircodota) [inline src/main/java/com/tabnine/binary/BinaryProcessGateway.kt:43]: maybe append to NO_PROXY env if one already defined?

**Review by amircodota — APPROVED** [review]
LGTM

File: src/main/java/com/tabnine/binary/BinaryProcessGateway.kt (hunk 2 above)
Comment (dimacodota) [reply src/main/java/com/tabnine/binary/BinaryProcessGateway.kt:43]: this is the case of `!useIJProxySettings` ie the user  explicitly selected to "not use the existing proxy settings"

**Pull request merged**